Tryouts/Prototypes/Shell/App.xaml.cs

```diff
@@ -43,5 +53,32 @@ private void StartWithWebWindowOptions(WebWindowOptions options)
             Application.Current.ShutdownMode = ShutdownMode.OnLastWindowClose;
             webWindow.Show();
         }
+
+        protected override async void OnStartup(StartupEventArgs e)
+        {
+            base.OnStartup(e);
+
+            var host = new HostBuilder()
+            .ConfigureAppConfiguration(
+                config => config.AddJsonFile("appsettings.json"))
+            .ConfigureLogging(l => l.AddDebug().SetMinimumLevel(LogLevel.Debug))
+            .ConfigureServices(
+                (context, services) => services
+                    .AddMessageRouterServer(mr => mr.UseWebSockets())
+                    .Configure<MessageRouterWebSocketServerOptions>(
+                        context.Configuration.GetSection("MessageRouter:WebSocket"))
+                    .Configure<LoggerFactoryOptions>(context.Configuration.GetSection("Logging")))
+            .Build();
+
+            await host.StartAsync(_cancellationTokenSource.Token);
+        }
+
+        protected override async void OnExit(ExitEventArgs e)
+        {
+            base.OnExit(e);
```

Comment:
This won't work properly, the host must be stopped and the app won't wait for the async operations to complete. I say we merge this as is so that we can integrate this work, and we can fix it in a new PR.